examples/large_models/inferentia2/util/inf2_save_split_checkpoints.py

```diff
@@ -41,13 +44,25 @@ def amp_callback(model: OPTForCausalLM, dtype: torch.dtype) -> None:
 
 save_path = create_directory_if_not_exists(args.save_path)
 
+# Load HuggingFace model config
+hf_model_config = AutoConfig.from_pretrained(args.model_name)
+
 # Load HuggingFace model
-hf_model = OPTForCausalLM.from_pretrained(args.model_name, low_cpu_mem_usage=True)
+hf_model = AutoModelForCausalLM.from_pretrained(args.model_name, low_cpu_mem_usage=True)
 
 # Apply Automatic Mixed Precision (AMP)
-amp_callback(hf_model, torch.float16)
+if hf_model_config.model_type == "opt":
+    opt_amp_callback(hf_model, torch.float16)
 
 # Save the model
 save_pretrained_split(hf_model, args.save_path)
 
+# Load and save tokenizer for the model
+if hf_model_config.model_type == "llama":
+    tokenizer = LlamaTokenizer.from_pretrained(args.model_name)
+else:
+    tokenizer = AutoTokenizer.from_pretrained(args.model_name)
```

Comment:
Will AutoTokenizer not load the LlamaTokenizer as well?

Reply:
Updated